// ng-modules/dist/components/modal/alert/alert.js
import { Component } from "@angular/core";
import { ModalController } from "../../../providers/modal/modalController";
var AlertComponent = (function () {
    function AlertComponent(modalController) {
        this.modalController = modalController;
        this.image = "";
        this.content = "";
    }
    AlertComponent.prototype.ngOnInit = function () {
        var options = this.modalController.params;
        if (typeof options == "string") {
            options = {
                content: options,
            };
        }
        this.deal(options);
    };
    AlertComponent.prototype.deal = function (options) {
        this.content = options.content;
        this.image = options.image;
        this.contentTpl = options.contentTpl;
    };
    AlertComponent.prototype.ngOnDestroy = function () { };
    return AlertComponent;
}());
export { AlertComponent };
AlertComponent.decorators = [
    { type: Component, args: [{
                template: "<div class=\"alert-container\"> <div class=\"alert-image\" *ngIf=\"image&&(!contentTpl)\"> <img [src]=\"image\"> </div> <div class=\"alert-content\" *ngIf=\"(!contentTpl)\" [innerHTML]=\"content\"></div> <div *ngIf=\"contentTpl\"> <ng-container *ngTemplateOutlet=\"contentTpl\"> </ng-container> </div> </div> ",
                styles: [".alert-container {\n  height: 100%;\n  padding-bottom: 10px;\n  position: relative;\n  display: flex;\n  flex-direction: column;\n}\n.alert-container .alert-image {\n  flex-grow: 0.7;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n}\n.alert-container .alert-image img {\n  max-width: 80%;\n  max-height: 90%;\n}\n.alert-container .alert-content {\n  overflow: auto;\n  margin: auto;\n  width: fit-content;\n  height: fit-content;\n  max-height: 100%;\n  max-width: 100%;\n  padding: 10px;\n}"],
            },] },
];
AlertComponent.ctorParameters = function () { return [
    { type: ModalController, },
]; };
