// ng-modules/dist/components/form/daterange/daterange.js
import { Directive, ElementRef, forwardRef } from "@angular/core";
import { NG_VALUE_ACCESSOR } from "@angular/forms";
import { Input } from "@angular/core";
import { UtilProvider } from "../../../providers/util/util";
import * as moment from "moment";
import { I18nProvider } from "../../../providers/i18n/i18n";
var DaterangeDirective = (function () {
    function DaterangeDirective(element, util) {
        this.element = element;
        this.util = util;
    }
    DaterangeDirective.prototype.writeValue = function (value) {
        var format = "YYYY-MM-DD";
        if (this.daterange && this.daterange.locale && this.daterange.locale.format) {
            format = this.daterange.locale.format;
        }
        if (value && value.match(" - ")) {
            var data = value.split(" - ");
            $(this.element.nativeElement).data("daterangepicker").setStartDate(new Date(data[0]));
            $(this.element.nativeElement).data("daterangepicker").setStartDate(new Date(data[1]));
            $(this.element.nativeElement).val(moment(data[0]).format(format) + " - " + moment(data[1]).format(format));
        }
        else {
            $(this.element.nativeElement).val("");
        }
    };
    DaterangeDirective.prototype._registerOnChange = function (data) { };
    DaterangeDirective.prototype.registerOnChange = function (fn) {
        this._registerOnChange = fn;
    };
    DaterangeDirective.prototype.touched = function () { };
    DaterangeDirective.prototype.registerOnTouched = function (fn) {
        this.touched = fn;
    };
    DaterangeDirective.prototype.setDisabledState = function (isDisabled) { };
    Object.defineProperty(DaterangeDirective.prototype, "daterange", {
        get: function () {
            return this._config;
        },
        set: function (value) {
            this._config = value;
            this.setting(value);
        },
        enumerable: true,
        configurable: true
    });
    ;
    DaterangeDirective.prototype.ngOnInit = function () {
    };
    DaterangeDirective.prototype.setting = function (config) {
        var data = this.util.extend({
            singleDatePicker: false,
            autoUpdateInput: false,
            locale: I18nProvider.get("dateSetting"),
        }, config, true);
        $(this.element.nativeElement).daterangepicker(data);
        var that = this;
        $(this.element.nativeElement).on('apply.daterangepicker', function (ev, picker) {
            var format = "YYYY-MM-DD";
            if (that.daterange && that.daterange.locale && that.daterange.locale.format) {
                format = that.daterange.locale.format;
            }
            that._registerOnChange(picker.startDate.toISOString() + " - " + picker.endDate.toISOString());
            $(this).val(picker.startDate.format(format) + " - " + picker.endDate.format(format));
        });
        $(this.element.nativeElement).on('cancel.daterangepicker', function (ev, picker) {
        });
    };
    return DaterangeDirective;
}());
export { DaterangeDirective };
DaterangeDirective.decorators = [
    { type: Directive, args: [{
                selector: "[daterange]",
                providers: [
                    { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(function () { return DaterangeDirective; }), multi: true }
                ],
            },] },
];
DaterangeDirective.ctorParameters = function () { return [
    { type: ElementRef, },
    { type: UtilProvider, },
]; };
DaterangeDirective.propDecorators = {
    'daterange': [{ type: Input, args: ["daterange",] },],
};
