// ng-modules/dist/pipes/se/se.js
import { Pipe } from "@angular/core";
import { DomSanitizer } from '@angular/platform-browser';
var SeUrl = (function () {
    function SeUrl(domSanitizer) {
        this.domSanitizer = domSanitizer;
    }
    SeUrl.prototype.transform = function (value) {
        return this.domSanitizer.bypassSecurityTrustUrl(value);
    };
    return SeUrl;
}());
export { SeUrl };
SeUrl.decorators = [
    { type: Pipe, args: [{
                name: "SeUrl"
            },] },
];
SeUrl.ctorParameters = function () { return [
    { type: DomSanitizer, },
]; };
var SeStyle = (function () {
    function SeStyle(domSanitizer) {
        this.domSanitizer = domSanitizer;
    }
    SeStyle.prototype.transform = function (value) {
        return this.domSanitizer.bypassSecurityTrustStyle(value);
    };
    return SeStyle;
}());
export { SeStyle };
SeStyle.decorators = [
    { type: Pipe, args: [{
                name: "SeStyle"
            },] },
];
SeStyle.ctorParameters = function () { return [
    { type: DomSanitizer, },
]; };
var SeHtml = (function () {
    function SeHtml(domSanitizer) {
        this.domSanitizer = domSanitizer;
    }
    SeHtml.prototype.transform = function (value) {
        return this.domSanitizer.bypassSecurityTrustHtml(value);
    };
    return SeHtml;
}());
export { SeHtml };
SeHtml.decorators = [
    { type: Pipe, args: [{
                name: "SeHtml"
            },] },
];
SeHtml.ctorParameters = function () { return [
    { type: DomSanitizer, },
]; };
