// ng-modules/src/components/modal/alert/alert.ts
import { Component, TemplateRef } from "@angular/core";
import { ActivatedRouteSnapshot, ActivatedRoute } from "@angular/router";
import { ModalController } from "../../../providers/modal/modalController";
import { ModalOptions } from "../modal";

export interface AlertOptions extends ModalOptions {
    title?: string;
    image?: string;
    content?: string;
    contentTpl?: TemplateRef<any>;
}

@Component({
    templateUrl: "./alert.html",
    styleUrls: ["./alert.scss"],
})
export class AlertComponent {
    constructor(
        private modalController: ModalController
    ) { }
    image: string = "";
    content: string = "";
    contentTpl: TemplateRef<any>;

    ngOnInit() {
        let options = this.modalController.params;
        if (typeof options == "string") {
            options = {
                content: options,
            }
        }
        this.deal(options);
    }

    deal(options: AlertOptions) {
        this.content = options.content;
        this.image = options.image;
        this.contentTpl = options.contentTpl;
    }

    ngOnDestroy() { }
}
