// ng-modules/dist/components/ueditor/ueditor.js
import { Component, ViewChild, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var UEditorComponent = (function () {
    function UEditorComponent() {
        this._registerOnChange = function (data) { };
        this.isInit = false;
        this.ueReady = false;
    }
    UEditorComponent.prototype.writeValue = function (obj) {
        var _this = this;
        this._defaultContent = obj;
        if (this.ue && this.ueReady) {
            setTimeout(function () {
                if (_this.isInit) {
                    return;
                }
                _this.isInit = true;
                _this.ue.setContent(obj);
            }, 200);
        }
    };
    UEditorComponent.prototype.registerOnChange = function (fn) {
        this._registerOnChange = fn;
    };
    UEditorComponent.prototype.registerOnTouched = function (fn) {
    };
    UEditorComponent.prototype.setDisabledState = function (isDisabled) {
    };
    UEditorComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (!window.UE) {
            return console.error("请先引入ueditor,详情请联系 ikrong.com");
        }
        for (var key in window.UE.instants) {
            window.UE.instants[key].destroy();
            delete window.UE.instants[key];
        }
        window.UE.delEditor(this.editor.nativeElement);
        this.ue = window.UE.getEditor(this.editor.nativeElement, {
            toolbars: [["paragraph", "fontfamily", "fontsize", "|", 'forecolor', 'backcolor', 'bold', 'italic', 'underline', 'strikethrough', "lineheight", 'superscript', 'subscript', "|", "indent", "justifyleft", "justifycenter", "justifyright", "justifyjustify", "|", "link", "unlink", "|", 'removeformat', 'formatmatch', 'autotypeset', 'blockquote', 'pasteplain', "|", "insertorderedlist", "insertunorderedlist", "rowspacingtop", "rowspacingbottom", "directionalityltr", "directionalityrtl", "touppercase", "pagebreak", "spechars", "|", "simpleupload", "imagenone", "imageleft", "imageright", "imagecenter", "|", "selectall", "cleardoc", "source"]]
        });
        this.ue.ready(function () {
            var height = _this.editor.nativeElement.offsetParent.clientHeight - _this.editor.nativeElement.querySelector(".edui-editor-toolbarbox").clientHeight;
            _this.ue.setHeight(height < 0 ? 400 : height);
            _this.ueReady = true;
            setTimeout(function () {
                if (_this.isInit) {
                    return;
                }
                _this.isInit = true;
                _this.ue.setContent(_this._defaultContent || "");
            }, 200);
            _this.ue.on("contentChange", function () {
                _this._registerOnChange(_this.ue.getContent());
            });
        });
    };
    return UEditorComponent;
}());
export { UEditorComponent };
UEditorComponent.decorators = [
    { type: Component, args: [{
                selector: 'ueditor',
                template: "<div #editor class=\"editor-content\"></div> ",
                styles: [""],
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(function () { return UEditorComponent; }),
                        multi: true
                    }
                ]
            },] },
];
UEditorComponent.ctorParameters = function () { return []; };
UEditorComponent.propDecorators = {
    'editor': [{ type: ViewChild, args: ["editor",] },],
};
