// ng-modules/dist/components/form/formInputStatus.js
import { Directive, ElementRef, Renderer2, Optional } from "@angular/core";
import { NgModel, NgControl } from "@angular/forms";
var FormInputStatus = (function () {
    function FormInputStatus(element, renderer2, ngModel, ngControl) {
        this.element = element;
        this.renderer2 = renderer2;
        this.ngModel = ngModel;
        this.ngControl = ngControl;
    }
    FormInputStatus.prototype.getParent = function (path) {
        var _this = this;
        if (!this.parentElement) {
            path.some(function (item) {
                if (item.nodeName) {
                    var node = item.nodeName.toLocaleLowerCase();
                    if (node == "label") {
                        _this.parentElement = item;
                        return true;
                    }
                }
            });
            return this.parentElement;
        }
        else {
            return this.parentElement;
        }
    };
    FormInputStatus.prototype.error = function () {
        if (this.parentElement) {
            this.renderer2.addClass(this.parentElement, "invalid");
            this.renderer2.removeClass(this.parentElement, "valid");
        }
        this.renderer2.addClass(this.element.nativeElement, "invalid");
        this.renderer2.removeClass(this.element.nativeElement, "valid");
    };
    FormInputStatus.prototype.success = function () {
        if (this.parentElement) {
            this.renderer2.addClass(this.parentElement, "valid");
            this.renderer2.removeClass(this.parentElement, "invalid");
        }
        this.renderer2.addClass(this.element.nativeElement, "valid");
        this.renderer2.removeClass(this.element.nativeElement, "invalid");
    };
    return FormInputStatus;
}());
export { FormInputStatus };
FormInputStatus.decorators = [
    { type: Directive, args: [{
                selector: "input,select,textarea",
            },] },
];
FormInputStatus.ctorParameters = function () { return [
    { type: ElementRef, },
    { type: Renderer2, },
    { type: NgModel, decorators: [{ type: Optional },] },
    { type: NgControl, decorators: [{ type: Optional },] },
]; };
