// ng-modules/src/module.ts
import { NgModule } from '@angular/core';
import { ListComponent, ListCellComponent, ModalComponent, ModalContentComponent, AlertComponent, MoveableComponent, FullscreenComponent, ConfirmComponent, FormInputStatus, SelectorComponent, LoadingComponent, ToastComponent, TabsComponent, TabContentDirective, TabSwitchDirective, UEditorComponent, DaterangeDirective, DatetimeDirective, InfoConfirmComponent, PopoverComponent } from './components';
import { ModalProvider, DomProvider, UtilProvider, AnimationProvider, I18nProvider, I18nProviderFactory, I18NToken, PopoverProvider, } from './providers';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { Datetime, SeHtml, SeStyle, SeUrl, I18NPipe } from './pipes';
import { RouterModule } from '@angular/router';
import { ErrorHandler } from '@angular/core/src/error_handler';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ModuleWithProviders } from '@angular/core/src/metadata/ng_module';

@NgModule({
    declarations: [
        ListComponent,
        ListCellComponent,
        ModalComponent,
        ModalContentComponent,
        Datetime,
        I18NPipe,
        SeHtml,
        SeStyle,
        SeUrl,
        AlertComponent,
        ConfirmComponent,
        InfoConfirmComponent,
        LoadingComponent,
        MoveableComponent,
        FullscreenComponent,
        FormInputStatus,
        SelectorComponent,
        ToastComponent,
        TabsComponent,
        TabContentDirective,
        TabSwitchDirective,
        UEditorComponent,
        DaterangeDirective,
        DatetimeDirective,
        PopoverComponent,
    ],
    entryComponents: [
        ListComponent,
        ModalComponent,
        AlertComponent,
        ConfirmComponent,
        InfoConfirmComponent,
        FullscreenComponent,
        LoadingComponent,
        ToastComponent,
        TabsComponent,
        UEditorComponent,
        PopoverComponent,
    ],
    exports: [
        ListComponent,
        ListCellComponent,
        ModalComponent,
        ModalContentComponent,
        Datetime,
        I18NPipe,
        SeHtml,
        SeStyle,
        SeUrl,
        AlertComponent,
        ConfirmComponent,
        InfoConfirmComponent,
        LoadingComponent,
        RouterModule,
        MoveableComponent,
        FullscreenComponent,
        FormInputStatus,
        SelectorComponent,
        ToastComponent,
        TabsComponent,
        TabContentDirective,
        TabSwitchDirective,
        UEditorComponent,
        DaterangeDirective,
        DatetimeDirective,
        PopoverComponent,
    ],
    imports: [
        CommonModule,
        FormsModule,
        RouterModule,
        ReactiveFormsModule,
    ],
})
export class CoreModule {

    constructor() { }

    static forRoot(config: CoreModuleConfig = {} as any): ModuleWithProviders {
        return {
            ngModule: CoreModule,
            providers: [
                ModalProvider,
                PopoverProvider,
                DomProvider,
                UtilProvider,
                AnimationProvider,
                { provide: I18NToken, useValue: config.i18n || {}, },
                { provide: I18nProvider, useFactory: I18nProviderFactory, deps: [I18NToken], },
            ]
        }
    }

}

export interface CoreModuleConfig {
    i18n?: {
        locales?: any[];
        default?: string,
    };
}
