// ng-modules/dist/components/popover/popover.js
import { Component, Input, TemplateRef, ViewChild, ElementRef, HostListener } from "@angular/core";
import { PopoverProvider } from "../../providers/modal/popover";
var PopoverComponent = (function () {
    function PopoverComponent(ele, popoverProvider) {
        this.ele = ele;
        this.popoverProvider = popoverProvider;
        this.position = "right";
    }
    Object.defineProperty(PopoverComponent.prototype, "popover", {
        set: function (value) {
            if (value instanceof TemplateRef) {
                this.tpl = value;
                this._popover = this.tpl;
            }
        },
        enumerable: true,
        configurable: true
    });
    ;
    PopoverComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.popCtrl = this.popoverProvider.create({
            content: this.popEle.nativeElement,
            position: this.position,
            width: this.width,
            height: this.height,
            parent: this.ele.nativeElement,
        });
        this.popEle.nativeElement.addEventListener("mouseover", function () {
            _this.popCtrl.show();
        });
        this.popEle.nativeElement.addEventListener("mouseout", function () {
            _this.popCtrl.hide();
        });
    };
    PopoverComponent.prototype.bindBodyClick = function () {
        var _this = this;
        document.body.addEventListener("click", this._fn = function (e) {
            if (!e.path.some(function (item) {
                if (item == _this.popoverOutbox.nativeElement || item == _this.parentEle) {
                    return true;
                }
            })) {
                _this.popCtrl.hide();
            }
        });
    };
    PopoverComponent.prototype.clearBodyClick = function () {
        document.body.removeEventListener("click", this._fn);
    };
    PopoverComponent.prototype.mouseOver = function (e) {
        this.popCtrl.show();
    };
    PopoverComponent.prototype.mouseOut = function (e) {
        this.popCtrl.hide();
    };
    PopoverComponent.prototype.getRect = function () {
        var pop = this.parentEle;
        return pop.getBoundingClientRect();
    };
    return PopoverComponent;
}());
export { PopoverComponent };
PopoverComponent.decorators = [
    { type: Component, args: [{
                selector: "[popover]",
                template: "<ng-content></ng-content> <div #popEle> <ng-container *ngTemplateOutlet=\"tpl\"></ng-container> </div> ",
                styles: [""]
            },] },
];
PopoverComponent.ctorParameters = function () { return [
    { type: ElementRef, },
    { type: PopoverProvider, },
]; };
PopoverComponent.propDecorators = {
    'popover': [{ type: Input, args: ["popover",] },],
    'popEle': [{ type: ViewChild, args: ["popEle",] },],
    'popoverOutbox': [{ type: ViewChild, args: ["popoverOutbox",] },],
    'width': [{ type: Input, args: ["popWidth",] },],
    'height': [{ type: Input, args: ["popHeight",] },],
    'position': [{ type: Input, args: ["popPosition",] },],
    'mouseOver': [{ type: HostListener, args: ['mouseover', ['$event'],] },],
    'mouseOut': [{ type: HostListener, args: ['mouseout', ['$event'],] },],
};
