// ng-modules/dist/components/fullscreen/fullscreen.js
import { Component, ElementRef } from '@angular/core';
var FullscreenComponent = (function () {
    function FullscreenComponent(elementRef) {
        this.elementRef = elementRef;
        this.distance = 10;
        this.widthTimestamp = 0;
        this.heightTimestamp = 0;
    }
    FullscreenComponent.prototype.getRate = function () {
        var element = this.elementRef.nativeElement;
        var parentScroll = element.offsetParent.scrollTop || window.scrollY;
        var height = this.height;
        var offsetTop = element.offsetTop;
        var cut = offsetTop - parentScroll;
        return {
            rate: cut / height,
            cut: cut,
            height: height,
            offsetTop: offsetTop,
        };
    };
    FullscreenComponent.prototype.ngOnInit = function () {
        var _this = this;
        var element = this.elementRef.nativeElement;
        var timer;
        element.addEventListener("mousewheel", function () {
            timer && clearTimeout(timer);
            timer = setTimeout(function () {
                var rate = _this.getRate();
                if (rate.rate < 0.2 && rate.rate > 0) {
                }
            }, 200);
        }, {
            passive: true,
        });
    };
    Object.defineProperty(FullscreenComponent.prototype, "width", {
        get: function () {
            if ((+new Date) - this.widthTimestamp > this.distance) {
                this.widthTimestamp = (+new Date);
                return this._width = window.innerWidth;
            }
            else {
                return this._width;
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FullscreenComponent.prototype, "height", {
        get: function () {
            if ((+new Date) - this.heightTimestamp > this.distance) {
                this.heightTimestamp = (+new Date);
                return this._height = window.innerHeight;
            }
            else {
                return this._height;
            }
        },
        enumerable: true,
        configurable: true
    });
    return FullscreenComponent;
}());
export { FullscreenComponent };
FullscreenComponent.decorators = [
    { type: Component, args: [{
                selector: "fullscreen",
                template: "<div class=\"fullscreen\" [style.height.px]=\"height\"> <ng-content></ng-content> </div> ",
                styles: [".fullscreen {\n  position: relative;\n  width: 100%;\n  overflow: hidden;\n}"],
            },] },
];
FullscreenComponent.ctorParameters = function () { return [
    { type: ElementRef, },
]; };
